crates/sui-json-rpc/src/read_api.rs

```diff
@@ -148,14 +149,14 @@ impl RpcReadApiServer for ReadApi {
     async fn get_dynamic_field_object(
         &self,
         parent_object_id: ObjectID,
-        name: String,
+        name: DynamicFieldName,
```

Comment:
We also need to update https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/providers/json-rpc-provider.ts#L943 . I am happy to work on this early next week